zrml/market-commons/src/migrations.rs

```diff
@@ -14,3 +14,288 @@
+            assert_eq!(updated_market.oracle, legacy_market.oracle);
+            assert_eq!(updated_market.metadata, legacy_market.metadata);
+            assert_eq!(updated_market.market_type, legacy_market.market_type);
+            assert_eq!(updated_market.period, legacy_market.period);
+            assert_eq!(updated_market.deadlines, legacy_market.deadlines);
+            assert_eq!(updated_market.scoring_rule, legacy_market.scoring_rule);
+            assert_eq!(updated_market.status, legacy_market.status);
+            assert_eq!(updated_market.report, legacy_market.report);
+            assert_eq!(updated_market.resolved_outcome, legacy_market.resolved_outcome);
+        }
+        Ok(())
+    }
+}
+
+mod utility {
+    use alloc::vec::Vec;
+    use frame_support::StorageHasher;
+    use parity_scale_codec::Encode;
+
+    #[allow(unused)]
+    pub fn key_to_hash<H, K>(key: K) -> Vec<u8>
+    where
+        H: StorageHasher,
+        K: Encode,
+    {
+        key.using_encoded(H::hash).as_ref().to_vec()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{market_commons_pallet_api::MarketCommonsPalletApi, mock::*};
+    use alloc::{vec, vec::Vec};
+    use core::fmt::Debug;
+    use frame_support::{Blake2_128Concat, StorageHasher};
+    use parity_scale_codec::Encode;
+    use zeitgeist_primitives::types::{
+        Deadlines, Market, MarketCreation, MarketDisputeMechanism, MarketId, MarketPeriod,
+        MarketStatus, MarketType,
+    };
+
+    #[test]
+    fn test_on_runtime_upgrade() {
+        ExtBuilder::default().build().execute_with(|| {
+            StorageVersion::new(MARKET_COMMONS_REQUIRED_STORAGE_VERSION).put::<Pallet<Runtime>>();
+            let (legacy_markets, expected_markets) = create_test_data_for_market_update();
+            populate_test_data::<Blake2_128Concat, MarketId, LegacyMarketOf<Runtime>>(
+                MARKET_COMMONS,
+                MARKETS,
+                legacy_markets,
+            );
+            UpdateMarketsForAuthorizedMDM::<Runtime>::on_runtime_upgrade();
+            assert_eq!(
+                StorageVersion::get::<Pallet<Runtime>>(),
+                MARKET_COMMONS_NEXT_STORAGE_VERSION
+            );
+            for (market_id, market_expected) in expected_markets.iter().enumerate() {
+                let market_actual = <crate::Pallet<Runtime> as MarketCommonsPalletApi>::market(
+                    &(market_id as u128),
+                )
+                .unwrap();
+                assert_eq!(market_actual, *market_expected);
+            }
+        });
+    }
+
+    fn create_test_data_for_market_update() -> (Vec<LegacyMarketOf<Runtime>>, Vec<MarketOf<Runtime>>)
+    {
+        let deadlines = Deadlines {
+            grace_period: 2_u32.into(),
+            oracle_duration: 2_u32.into(),
+            dispute_duration: 2_u32.into(),
+        };
+        let old_markets: Vec<LegacyMarketOf<Runtime>> = vec![
```

Comment:
The difference between these two test cases is that one is categorical, while one is scalar. Woudn't it me more useful to have one market that has `Authorized` as MDM and one that has, say, `Court`, so that we can check if `Court` is correctly changed. I mean, the migration doesn't care if the market is categorical or scalar, but it does care very much what the dispute mechanism is.